File: app/src/test/java/com/example/david/lists/view/userlistlist/UserListDiffCallbackTest.java
package com.example.david.lists.view.userlistlist;

import com.example.david.lists.data.datamodel.UserList;

import org.junit.Test;

import static org.hamcrest.Matchers.is;
import static org.junit.Assert.assertThat;

public class UserListDiffCallbackTest {

    private String oneId = "id_one";
    private String oneTitle = "title_one";
    private int onePosition = 1;
    private UserList one = new UserList(oneId, new UserList(oneTitle, onePosition));

    private String twoId = "id_two";
    private String twoTitle = "title_two";
    private int twoPosition = 2;
    private UserList two = new UserList(twoId, new UserList(twoTitle, twoPosition));

    private UserListDiffCallback diffCallback = new UserListDiffCallback();


    @Test
    public void areItemsTheSameTrue() {
        assertThat(
                diffCallback.areItemsTheSame(one, one),
                is(true)
        );
    }

    @Test
    public void areItemsTheSameFalse() {
        assertThat(
                diffCallback.areItemsTheSame(one, two),
                is(false)
        );
    }


    @Test
    public void areContentsTheSameTrue() {
        assertThat(
                diffCallback.areContentsTheSame(one, one),
                is(true)
        );
    }

    @Test
    public void areContentsTheSameFalse() {
        assertThat(
                diffCallback.areContentsTheSame(one, two),
                is(false)
        );
    }
}

File: app/src/test/java/com/example/david/lists/view/userlistlist/UserListDiffCallbackTest.kt
package com.example.david.lists.view.userlistlist

import com.example.domain.datamodel.UserList
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class UserListDiffCallbackTest {

    private val one = UserList("title_one", 1, "id_one")
    private val two = UserList("title_two", 2, "id_two")

    private val diffCallback = UserListDiffCallback()


    @Test
    fun `areItemsTheSame - True`() {
        assertThat(diffCallback.areItemsTheSame(one, one)).isTrue()
    }

    @Test
    fun `areItemsTheSame - False`() {
        assertThat(diffCallback.areItemsTheSame(one, two)).isFalse()
    }


    @Test
    fun `areContentsTheSame - True`() {
        assertThat(diffCallback.areContentsTheSame(one, one)).isTrue()
    }

    @Test
    fun `areContentsTheSame - False`() {
        assertThat(diffCallback.areContentsTheSame(one, two)).isFalse()
    }
}